### PR TITLE
D3 transition from v3 to v4

### PR DESCRIPTION
**Pull request by korenkovv**

### CODE REVIEW

File: WebClient/doc/components/timeSlider/timeSlider.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+Message payload is an object where each property name represents an action to be done while property value represents action's data or settings. All properties are arbitrary.
+
+| Property | Type | Description |
+|---|---|---|
+| setEvents | Array &lt;[Event](#event)&gt; | Sets events array to be shown on the time slider. If there were already events shown then they will be replaced with these new ones. |
+| addEvents | Array &lt;[Event](#event)&gt; | Extends current time slider events array with additional events. All events will be shown on the time slider. |
+| setCurrentTime | string  | Sets the current time. The value of this property is a string in following [time format](#timeFormat). Current time will be immediately displayed in the time slider and time slider's scale will be centered at this time. The zoom level of the time slider will remain. |
+| setBrush | [Brush](#brush) | Selects a time range. If brush feature is enabled then corresponding time range will be displayed on the time slider as [d3-brush](https://github.com/d3/d3-brush) object. *Brush* can either be `{ }` or a [Brush](#brush) object. `{ }` means that there is no brush. |
```
Comment (korenkovv) [inline WebClient/doc/components/timeSlider/timeSlider.md]: Here is an error: setBrush has a different structure. 
setBrush = { extent: [start, end] }

File: WebClient/doc/components/timeSlider/timeSlider.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+
+#### Client message payload <a name="clientMessagePayload"></a>
+Payload is an object which can contain one of the following properties.
+
+| Property | Type | Description |
+|---|---|---|
+| active | boolean | Indicates whether *time slider* is shown. `true` when shown, `false` when hidden. This message is sent whenever time slider visibility changes, e.g. when time slider is collapsed, expanded or removed.|
+| selectedTime | string | Currently selected time as a string of the following [time format](#timeFormat). This message is sent each time the current time changes. |
+| brush | [Brush](#brush) | Current time range selection. *Brush* can either be an empty object `{ }` or a [Brush](#brush) object. `{ }` means that there is no brush. This message is sent each time the current brush changes. |
```
Comment (korenkovv) [inline WebClient/doc/components/timeSlider/timeSlider.md]: brush should have a different structure: brush = { extent: [start, end] }

File: WebClient/Scripts/components/timeSlider/timeSliderController.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            var events = messagePayload.addEvents.map(convertEvent);
+            this.model.addEvents(events);
+        }
+
+        if (messagePayload.setCurrentTime) {
+            this.model.value = timeParser(messagePayload.setCurrentTime);
+        }
+
+        if (messagePayload.setBrush) {
+            if (messagePayload.setBrush.start && messagePayload.setBrush.end) {
```
Comment (korenkovv) [inline WebClient/Scripts/components/timeSlider/timeSliderController.js]: For consistency, brush should have the same structure as when sending a brush message to server: 
brush = { extent: [start, end] }

**Review by cornelissenja — CHANGES_REQUESTED** [review]
Please add an extra field for the events that can be set from the publisher and will we returned over the websocket to the publisher when selected. I suggest something like "tag" and it would be nice if it can contain an arbitrary json structure so it is flexible to pass all information the publisher might need when returned.

**Review by cornelissenja — CHANGES_REQUESTED** [review]
In docs it says weather (climat) but i think you mean whether

**Review by cornelissenja — COMMENTED** [review]
Rename d3.css it now looks like it is the default css file for d3; maybe d3xxxx.css, or maybe move it to common?

File: WebClient/Scripts/components/timeSlider/timeSliderController.js (hunk 3 above)
Comment (cornelissenja) [reply WebClient/Scripts/components/timeSlider/timeSliderController.js]: Good idea, have to fix the publishers then thou

Comment (korenkovv) [reply WebClient/Scripts/components/timeSlider/timeSliderController.js]: This is a new message, previously we didn't have it. Therefore I think publisher doesn't send a brush message and we don't need to fix it there.

**Review by cornelissenja — COMMENTED** [review]
Looks very clean, maybe use your style as a guideline?

**Review by cornelissenja — COMMENTED** [review]
finished commenting :-)

**Pull request closed without merging**